lib/src/acter_avatar.dart

```diff
@@ -162,245 +126,242 @@ class _ActerAvatar extends State<ActerAvatar> {
+    switch (avatar.tooltip) {
+      case TooltipStyle.DisplayName:
+        return Tooltip(
+          message: avatar.displayName ?? avatar.uniqueId,
+          child: child,
+        );
+      case TooltipStyle.UniqueId:
+        return Tooltip(message: avatar.uniqueId, child: child);
+      case TooltipStyle.Combined:
+        var message = avatar.uniqueName ?? avatar.uniqueId;
+        if (avatar.displayName != null) {
+          message = '${avatar.displayName} (${avatar.uniqueId})';
+        }
+        return Tooltip(message: message, child: child);
+      case TooltipStyle.None:
+        return child;
     }
   }
 
   Widget renderWithAvatar(BuildContext context, ImageProvider avatar) {
-    switch (widget.mode) {
-      case DisplayMode.DM:
-        return CircleAvatar(
-          foregroundImage: avatar,
-          onForegroundImageError: avatarError,
-          radius: widget.size ?? 24,
-        );
-      case DisplayMode.Space:
-        return Stack(
-          clipBehavior: Clip.none,
-          children: <Widget>[
-            Container(
-              height: widget.size ?? 48,
-              width: widget.size ?? 48,
-              decoration: BoxDecoration(
-                borderRadius: BorderRadius.circular(6.0),
-                image: DecorationImage(
-                  fit: BoxFit.cover,
-                  image: avatar,
-                  onError: avatarError,
+    return widget.options.when(
+        // default render
+        (avatarInfo, parentBadges, size, badgesSize) => Stack(
+              clipBehavior: Clip.none,
+              children: <Widget>[
+                Container(
+                  height: size ?? 48,
+                  width: size ?? 48,
+                  decoration: BoxDecoration(
+                    borderRadius: BorderRadius.circular(6.0),
+                    image: DecorationImage(
+                      fit: BoxFit.cover,
+                      image: avatar,
+                      onError: avatarError,
+                    ),
+                  ),
                 ),
-              ),
+                // if badges specified, render it.
+                if (parentBadges != null && parentBadges.isNotEmpty)
+                  renderBadges(context, widget.options as DefaultAvatarOptions,
+                      badgesSize),
+              ],
             ),
-            renderSpaceParent(context),
-          ],
-        );
-      case DisplayMode.GroupChat:
-        return Container(
-          height: widget.size ?? 48,
-          width: widget.size ?? 48,
-          decoration: BoxDecoration(
-            borderRadius: BorderRadius.circular(6.0),
-            image: DecorationImage(
-              fit: BoxFit.cover,
-              image: avatar,
-              onError: avatarError,
+        // render for DM
+        DM: (avatarInfo, size) => CircleAvatar(
+              foregroundImage: avatar,
+              onForegroundImageError: avatarError,
+              radius: size ?? 24,
```

Comment:
the diffs here make it hard to follow what has actually changed. I glanced over these and tried to find difference but failed to do so. If there is anything substantively different I should be looking it, please point it out and I will.

Reply:
Yeah, the code is a bit complex to follow. I'll point out areas where it is meaningful to get understanding of whole picture 👍🏻 !

Reply:
I understand the picture overall, thanks for the pointers. But what I meant was if there was any changes in the way that things are rendered compared to before ... I dunno changes of sizes or stacking order or anything like that ... that is hard to find out.